rotkehlchen/assets/utils.py

```diff
@@ -94,27 +175,20 @@ def get_or_create_evm_token(
             )
 
             if evm_inquirer is not None:
-                if token_kind == EvmTokenKind.ERC20:
-                    info = evm_inquirer.get_erc20_contract_info(evm_address)
-                    decimals = info['decimals'] if decimals is None else decimals
-                    symbol = info['symbol'] if symbol is None else symbol
-                    name = info['name'] if name is None else name
-                    if None in (decimals, symbol, name):
-                        raise NotERC20Conformant(f'Token {evm_address} is not ERC20 conformant')  # noqa: E501 B904  # pylint: disable=raise-missing-from
-
-                elif token_kind == EvmTokenKind.ERC721:
-                    info = evm_inquirer.get_erc721_contract_info(evm_address)
-                    decimals = 0
-                    if symbol is None:
-                        symbol = info['symbol'] if info['symbol'] is not None else ''
-                    if name is None:
-                        name = info['name'] if info['name'] is not None else ''
-
-                else:
-                    raise NotERC20Conformant(f'Token {evm_address} is of uknown type')  # pylint: disable=raise-missing-from  # noqa: E501
+                name, symbol, decimals = _query_token_information(
+                    evm_inquirer=evm_inquirer,
+                    evm_address=evm_address,
+                    name=name,
+                    symbol=symbol,
+                    decimals=decimals,
+                    token_kind=token_kind,
+                )
+            # make sure that basic information is always filled
+            name = identifier if name is None else name
+            decimals = 18 if decimals is None else decimals
```

Comment:
unless I read the code wrong as I said the type being wrong in `_query_token_information` means that it should never return None So both of these lines are not needed.

```suggestion
```

Reply:
It can happen that None is not provided and the contract doesn't have the name field for example